src/App.tsx

```diff
@@ -1,26 +1,320 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import {
+  getTodos,
+  USER_ID,
+  createTodo,
+  deleteTodo,
+  updateTodo,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { Notification } from './components/Notification/Notification';
+import { Footer } from './components/Footer/Footer';
+import { NewTodo } from './components/NewTodo/NewTodo';
+import classNames from 'classnames';
+import { ErrorMessages } from './constants/ErrorMessages';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string>('');
+  const [filter, setFilter] = useState('all');
+  const [loading, setLoading] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [inputValue, setInputValue] = useState('');
+  const [todosToProcess, setTodosToProcess] = useState<Set<number>>(new Set());
```

Comment:
```suggestion
  const [todosToProcessIds, setTodosToProcessIds] = useState<Set<number>>(new Set());
```

if we're talking about `Set<number>()`, there should be ids